### PR TITLE
 feat(mqtt): split broker_url in multiple fields

### LINKED ISSUE #389: MQTT config for clientID
There is no option to define the clientId for the broker connection.
Please make this possible

### LINKED ISSUE #509: MQTT password visible in WebUI
## Describe the bug
After upgrading to the latest valetudo release we are able to configure mqtt from the UI. While this is pretty cool in general, I'm not really sure if it is a good idea to show the connection credentials/password in clear text to everyone who could access the web frontend... Would be good to have the ability to disable web editing of the mqtt part completely and/or split out the connection URL into different fields without displaying the current password (as it is done for the wifi config site).

### Valetudo Version
0.5.1

### PR DESCRIPTION
**Pull request by alexkn**
and hide passwords

resolves #509 
resolves #389

### CODE REVIEW

File: lib/MqttClient.js [hunk 1]
```diff
@@ -136,10 +155,12 @@ class MqttClient {
...
-            if (this.caPath) {
-                options.ca = fs.readFileSync(this.caPath);
-            }
-            this.client = mqtt.connect(this.brokerURL, options);
+            if (this.clientId) options.clientId = this.clientId;
+            if (this.username) options.username = this.username;
+            if (this.password) options.password = this.password;
+            if (this.caPath) options.ca = fs.readFileSync(this.caPath);
+
+            this.client = mqtt.connect((this.usetls ? "mqtts://" : "mqtt://") + this.server + ":" + this.port, options);
```
Comment (rumpeltux) [inline lib/MqttClient.js:163]: (nitpick) wrap this line

File: lib/webserver/ApiRouter.js [hunk 2]
```diff
@@ -714,11 +716,22 @@ const ApiRouter = function(webserver) {
...
+        res.json(mqttConfig);
     });
 
     router.put("/mqtt_config", async (req, res) => {
-        configuration.set("mqtt", req.body);
+        let mqttConfig = req.body;
+        let oldConfig = configuration.get("mqtt");
+
+        // keep password if not changed
+        if (oldConfig.server == mqttConfig.server && mqttConfig.password == "****") mqttConfig.password = oldConfig.password;
```
Comment (rumpeltux) [inline lib/webserver/ApiRouter.js:732]: this one too

File: lib/MqttClient.js (hunk 1 above)
Comment (alexkn) [reply lib/MqttClient.js:163]: How long should be the maximum length?
The if statement 7 lines above is also that long.

Maybe we should configure that in eslint.

Comment (alexkn) [reply lib/MqttClient.js]: fixed that line

File: lib/webserver/ApiRouter.js (hunk 2 above)
Comment (alexkn) [reply lib/webserver/ApiRouter.js]: done

File: lib/webserver/ApiRouter.js [hunk 3]
```diff
@@ -714,11 +716,24 @@ const ApiRouter = function(webserver) {
...
+        res.json(mqttConfig);
     });
 
     router.put("/mqtt_config", async (req, res) => {
-        configuration.set("mqtt", req.body);
+        let mqttConfig = req.body;
+        let oldConfig = configuration.get("mqtt");
+
+        // keep password if not changed
+        if (oldConfig.server == mqttConfig.server && mqttConfig.password == "****") {
```
Comment (Hypfer) [inline lib/webserver/ApiRouter.js:732]: too bad for people using `****` as their password

**Pull request merged**